### PR TITLE
Improve test coverage, add py37 checks, backward capability with py27

### PR DESCRIPTION
**Pull request by zifter**

### CODE REVIEW

File: qrcode/image/base.py [hunk 1]
```diff
@@ -12,13 +12,13 @@ def __init__(self, border, width, box_size, *args, **kwargs):
         self.pixel_size = (self.width + self.border*2) * self.box_size
         self._img = self.new_image(**kwargs)
 
-    def drawrect(self, row, col):
+    def drawrect(self, row, col):  # pragma: no cover
```
Comment (SmileyChris) [inline qrcode/image/base.py]: Let's make this an `@abc.abstractmethod` instead -- we don't need the comment, and it doesn't need a return value or `raise` line

File: qrcode/image/base.py [hunk 2]
```diff
@@ -12,13 +12,13 @@ def __init__(self, border, width, box_size, *args, **kwargs):
...
         """
         Draw a single rectangle of the QR code.
         """
         raise NotImplementedError("BaseImage.drawrect")
 
-    def save(self, stream, kind=None):
+    def save(self, stream, kind=None):  # pragma: no cover
```
Comment (SmileyChris) [inline qrcode/image/base.py]: Make this an `@abc.abstractmethod` too

File: qrcode/image/base.py [hunk 3]
```diff
@@ -37,7 +37,7 @@ def new_image(self, **kwargs):  # pragma: no cover
         """
         Build the image class. Subclasses should return the class created.
         """
-        return None
+        raise NotImplementedError("BaseImage.new_image")
```
Comment (SmileyChris) [inline qrcode/image/base.py]: This can also be an `@abc.abstractmethod`

**Comment by zifter** [comment]
@SmileyChris, hi!
I've fixed review notes, is it possible to merge it?

**Pull request merged**